sql/catalyst/src/main/scala/org/apache/spark/sql/types/Decimal.scala

```diff
@@ -420,7 +420,11 @@ final class Decimal extends Ordered[Decimal] with Serializable {
       // have overflowed our Long; in either case we must rescale dv to the new scale.
       dv = dv.setScale(scale, roundMode)
       if (dv.precision > precision) {
```

Comment:
what is the precision of `BigDecimal` in this case?

Reply:
it's 1, can see the docs in `java.math.BigDecimal`
<img width="650" alt="image" src="https://user-images.githubusercontent.com/12025282/203693808-37715386-1fda-4134-aafc-b558ee55b7fe.png">

Reply:
I think this makes sense. If we write `SELECT 0bd` in Spark, the returned decimal is also `decimal(1, 0)`. Maybe forbidding `decimal(0, 0)` is a better choice. What do you think @srielau ?